File: solutions/1786-count-the-number-of-consistent-strings/solution.js
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function(allowed, words) {
    const s = new Set(allowed);
    return words.filter(el => !el.split("").some(c => !s.has(c))).length;


    
};

File: solutions/1786-count-the-number-of-consistent-strings/solution.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        s = set()
        for char in allowed:
            s.add(char)
            
        count = 0
        
        def consistent(word):
            for char in word:
                if char not in s:
                    return False
                
            return True
        
        for word in words:
            if consistent(word):
                count += 1
            
        return count
